src/Search/SearchResults.php

```diff
@@ -205,6 +241,43 @@ private function getFacetInfo(string $facet, array $facetAgregation): ?array
         }
     }
 
+    /**
+     * Get combined research group aggregation values.
+     *
+     * @param $aggregations
+     *
+     * @return array
+     */
+    private function combineBuckets($aggregations, string $datasetBucketName, string $infoProductBucketName): array
+    {
+        $datasetBucket = [];
+        $datasetAggregations = $this->findKey($aggregations, $datasetBucketName);
+        if (array_key_exists('buckets', $datasetAggregations)) {
+            $datasetBucket = array_column(
+                $datasetAggregations['buckets'],
+                'doc_count',
+                'key'
+            );
+        }
+
+        $infoProductBucket = [];
+        $infoProductAggregations = $this->findKey($aggregations, $infoProductBucketName);
+        if (array_key_exists('buckets', $infoProductAggregations)) {
+            $infoProductBucket = array_column(
+                $infoProductAggregations['buckets'],
+                'doc_count',
+                'key'
+            );
+        }
+
+        $combinedBuckets = array();
+        foreach (array_keys($datasetBucket + $infoProductBucket) as $key) {
+            $combinedBuckets[$key] = (isset($datasetBucket[$key]) ? $datasetBucket[$key] : 0) + (isset($infoProductBucket[$key]) ? $infoProductBucket[$key] : 0);
```

Comment:
```suggestion
            $combinedBuckets[$key] = ($datasetBucket[$key] ?? 0) + ($infoProductBucket[$key] ?? 0);
```

Reply:
I didn't realize $null[$key] ?? 0 won't cause a 'Trying to access array offset on value of type null', where $null[$key] of course  will.